plugins/mumble_plugin.h

```diff
@@ -27,65 +26,30 @@
 # error No MUMBLE_PLUGIN_EXPORT definition available
 #endif
 
-// Visual Studio 2008 x86
-#if _MSC_VER == 1500 && defined(_M_IX86)
-# define MUMBLE_PLUGIN_MAGIC     0xd63ab7ef
-# define MUMBLE_PLUGIN_MAGIC_2   0xd63ab7fe
-# define MUMBLE_PLUGIN_MAGIC_QT  0xd63ab7ee
-// Visual Studio 2010 x86
-#elif _MSC_VER == 1600 && defined(_M_IX86)
-# define MUMBLE_PLUGIN_MAGIC    0xd63ab7f0
-# define MUMBLE_PLUGIN_MAGIC_2  0xd63ab7ff
-# define MUMBLE_PLUGIN_MAGIC_QT 0xd63ab70f
-// Visual Studio 2013 x86
-#elif _MSC_VER == 1800 && defined(_M_IX86)
-# define MUMBLE_PLUGIN_MAGIC    0xd63ab7c0
-# define MUMBLE_PLUGIN_MAGIC_2  0xd63ab7cf
-# define MUMBLE_PLUGIN_MAGIC_QT 0xd63ab7ca
-// Visual Studio 2013 x64
-#elif _MSC_VER == 1800 && defined(_M_X64)
-# define MUMBLE_PLUGIN_MAGIC    0x9f3ed4c0
-# define MUMBLE_PLUGIN_MAGIC_2  0x9f3ed4cf
-# define MUMBLE_PLUGIN_MAGIC_QT 0x9f3ed4ca
-// Visual Studio 2015 x86
-#elif _MSC_VER == 1900 && defined(_M_IX86)
-# define MUMBLE_PLUGIN_MAGIC    0xa9b8c7c0
-# define MUMBLE_PLUGIN_MAGIC_2  0xa9b8c7cf
-# define MUMBLE_PLUGIN_MAGIC_QT 0xa9b8c7ca
-// Visual Studio 2015 x64
-#elif _MSC_VER == 1900 && defined(_M_X64)
-# define MUMBLE_PLUGIN_MAGIC    0x1f2e3dc0
-# define MUMBLE_PLUGIN_MAGIC_2  0x1f2e3dcf
-# define MUMBLE_PLUGIN_MAGIC_QT 0x1f2e3dca
-// Generic plugin magic values for platforms
-// where we do not officially plugins other
-// than "link".
-#else
-# define MUMBLE_PLUGIN_MAGIC    0xf4573570
-# define MUMBLE_PLUGIN_MAGIC_2  0xf457357f
+# define MUMBLE_PLUGIN_MAGIC    0xf4573580
 # define MUMBLE_PLUGIN_MAGIC_QT 0xf457357a
-#endif
 
-#define MUMBLE_PLUGIN_VERSION 2
+#define MUMBLE_PLUGIN_VERSION 3
+
+#define MUMBLE_PID_LOOKUP_EOF  0
+#define MUMBLE_PID_LOOKUP_OK   1
+
+typedef void * MumblePIDLookupContext;
+typedef int MumblePIDLookupStatus;
+typedef MumblePIDLookupStatus (*MumblePIDLookup)(const MumblePIDLookupContext ctx, const wchar_t *str, unsigned long long *pid);
 
 typedef struct _MumblePlugin {
 	unsigned int magic;
-	const std::wstring &description;
-	const std::wstring &shortname;
-	void (MUMBLE_PLUGIN_CALLING_CONVENTION *about)(void *);
-	void (MUMBLE_PLUGIN_CALLING_CONVENTION *config)(void *);
-	int (MUMBLE_PLUGIN_CALLING_CONVENTION *trylock)();
+	unsigned int version;
```

Comment:
Before we had no function to query the version and just put it into the the other fields. Considering our plugins usually referred to the game version they were targeting as well as their own version this seemed to work out ok. What are the plans with this integer?

Reply:
I plan to use it to determine which is the newest plugin between the one in the system folder and the one in the user folder.

Reply:
We should try to figure out whether we can think of a way to get around that if possible. Having to hand-increment that counter every time we touch a plugin is error prone.

I assume the problem is that old auto-updated plugins might still be in the user folder when the user installed something with newer plugins? One way to handle this would be to only ship plugins via auto-update to begin with. As you mentioned they outdate quickly anyways so having them in the installer might not be ideal anyways. Maybe we can come up with other options...

(from a technical POV: To evaluate the version we have to load the plugin from different locations to compare. Is that easily done with the way it's currently setup?)

Reply:
Yes, the problem is that the old plugins may remain in the user folder and be loaded instead of the updated ones in the system folder.

Yes, it should be pretty easy. We just need to load the library from the system directory, the one from the user directory and then check the version variable.

Reply:
Guess we won't get around having such a version long as we want to ship them in the installer. Will be a constant minor PITA when updating / getting update PRs.

Reply:
How about just changing file's last change date?
Sure it's not 100% reliable. But really how many people touch a old outdated plugin, after the new one that they are using is already installed.